packagedependencies.txt

```diff
@@ -1,4 +1,4 @@
 eigen|3.4.0|eigen3|conan|conan-center|na|MPL2_only=True
 spdlog|0.14.0|spdlog|thirdParties@github|https://github.com/SolarFramework/binaries/releases/download|shared|
 nlohmann_json|3.9.1|nlohmann_json|conan|conan-center|na|
-boost#stable|1.84.0|boost|conan|conan-center|default|
+boost#stable|1.84.0|boost|conan|conan-center|static|
```

Comment:
Just wanted to mention that having boost already statically linked with XPCF, and adding it here as well could potentially lead to some problems. Even if we ensure there is no version conflict, it increases binary size, memory consumption, issues relative to static initialization order, ...
In an ideal case, we should have either:
- boost as a private dependency of XPCF (i.e. invisible to users of XPCF) and XPCF exposes features via its own API (as done with share ptr)
- if direct access to boost is required by XPCF and other components, this is a suitable case for makeing boost a shared library (it is indeed shared)
- if we really want to statically link XPCF with boost and still be able to use it in XPCF client code, we could try to prevent code pruning with options like [gcc](https://ftp.gnu.org/old-gnu/Manuals/ld-2.9.1/html_node/ld_3.html)'s `--whole-archive` or [Windows MSVC's](https://learn.microsoft.com/en-us/cpp/build/reference/wholearchive-include-all-library-object-files?view=msvc-170) `/WHOLEARCHIVE`

Reply:
The SolARFramework does direct calls to boost which are not exposed by XPCF. The --whole-archive will embed all boost symbols in the XPCF library, resulting in larger files. By adding boost when directly required by the current project, we just provide the possibility to add missing symbols. But you are right, if there is a misalignment of boost version, we will face conflicts.

Reply:
OK for merging.